### ISSUE TITLE
All the tests are positives even if there are violations.

### ISSUE THREAD

**Issue by dancol93** [dequelabs/axe-core#1639]
Expectation: Using axe-core with Karma and Mocha and not to get false positives.

Actual: When I execute the tests, axe.run() returns false positives even when the expect(results.violations.length).to.equal(0) should throw a test error, I tried to take the result from the function and it didn't work, even though the logs indicate there are accessibility violations, here's an example:

```js
axe.run(element, (err, results) => {
    if (err) throw err;
    expect(err).to.be.null;
    expect(results.violations).to.have.lengthOf(0);
});
```

![image](https://user-images.githubusercontent.com/16827815/59666336-ebef9a80-91b4-11e9-8548-9d643510dc42.png)

<pre><code>axe-core version: 3.2.2

Browser and Assistive Technology versions

For Tooling issues:
- Node version: 10.16.0
- Platform:  Windows</code></pre>

**Comment by straker** [comment]
Thanks for reporting the issue. I looked into this and it'll be a lot of work in order for the callback to throw an error properly (right now it eats the error). As a workaround, you can wrap the asserts in a try/catch and then call the `done` method with an error if one of them throws.

```js
it('should fail', (done) => {
  axe.run(element, (err, results) => {
    if (err) done(err);

    try {
      expect(err).to.be.null;
      expect(results.violations).to.have.lengthOf(0);
      done();
    } catch (e) {
      done(e);
    }
  });
});
```

**Comment by dylanb** [comment]
@straker is this a regression? What line of code is swallowing the error?

**Comment by straker** [comment]
@dylanb Everything in the `axe.run` code path is surrounded in try/catch that catch the error and log it. Even if we removed the error swallowing, all we are doing with the error is rejecting the promise which does nothing unless you're using es6 Promises on the `axe.run`.

https://github.com/dequelabs/axe-core/blob/develop/lib/core/public/run.js#L114-L121
https://github.com/dequelabs/axe-core/blob/develop/lib/core/public/run-rules.js#L79-L84

**Comment by dancol93** [comment]
Hi guys, any updates about this error?

**Comment by dylanb** [comment]
@dancol93 did you close this because the workaround was good enough?
